src/memory.rs

```diff
@@ -524,44 +538,44 @@ const BLOCK_SIZE: usize = 64;
 #[derive(Clone, Debug)]
 pub struct UndefMask {
```

Comment:
Hmm, well, this is a lot of code to look through, but here in particular I'm wondering... what is the gain of using `Size` in the below `UndefMask` code? It makes it bigger and a little harder to read, and I'm not sure I see the benefit.

I had similar feelings with the rest of the code, but it makes more sense to me with the layout-related code.